lib/OpenQA/Worker/Job.pm

```diff
@@ -568,6 +569,20 @@ sub _format_reason {
 sub _set_job_done {
     my ($self, $reason, $params, $callback) = @_;
 
+    # Retrieve extended reason if available
+    my $pooldir    = $self->worker->pool_directory;
+    my $state_file = path($pooldir)->child(BASE_STATEFILE);
+    try {
+        if (-e $state_file) {
```

Comment:
I still wonder. You are combining try/catch with an explicit if. I wonder if we can either take the "loop before you leap" or "it is better to ask for forgiveness than for permission" for both checks. I guess because we can't decide if the JSON is valid before trying to parse it then we can handle the file not being existant in catch as well?

Reply:
The explicit `if` is to handle old os-autoinst versions which don't write the file so we don't see "Corrupted state file ...." in that case.

> then we can handle the file not being existant in catch as well

In general this makes sense. But you know that Perl has no nice way to distinguish errors. I rather have the `-e` check instead of an if with regex matching in the catch.